go-api/utils/utils.go

```diff
@@ -0,0 +1,209 @@
+package utils
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+
+	"golang.org/x/crypto/scrypt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/cors"
+	"github.com/gofiber/fiber/v2/middleware/logger"
+	"github.com/jackc/pgx/v5"
+	"github.com/redis/go-redis/v9"
+)
+
+func GetPgx(c *fiber.Ctx) pgx.Conn {
+	return c.Locals("pgxConn").(pgx.Conn)
+}
+
+func GetRdb(c *fiber.Ctx) redis.Conn {
+	return c.Locals("rdb").(redis.Conn)
+}
+
+var ctx = context.Background()
+
+func SetRedis(c *fiber.Ctx, key string, value string) error {
+	redisConn := GetRdb(c)
+
+	_err := redisConn.Set(c.Context(), key, value, 0).Err()
+	if _err != nil {
+		return _err
+	}
+	return nil
+}
+
+func GetRedis(c *fiber.Ctx, key string) (string, error) {
+	redisConn := GetRdb(c)
+	val, _err := redisConn.Get(c.Context(), key).Result()
+	if _err != nil {
+		return "", _err
+	}
+	return val, nil
+}
+
+func RawQueryWithoutReturn(c *fiber.Ctx, query string, args map[string]any) error {
+	pgxConn := GetPgx(c)
+	rows, _err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if _err != nil {
+		return _err
+	}
+	rows.Close()
+
+	return nil
+}
+
+func QueryRow[T any](c *fiber.Ctx, query string, args map[string]any) (T, error) {
+	var result T
+	pgxConn := GetPgx(c)
+
+	rows, _err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if _err != nil {
+		return result, _err
+	}
+
+	result, _err = pgx.CollectExactlyOneRow(rows, pgx.RowToStructByName[T])
+	if _err != nil {
+		return result, _err
+	}
+
+	return result, nil
+}
+
+func QueryRows[T any](c *fiber.Ctx, query string, args map[string]any) ([]T, error) {
+	results := []T{}
+	pgxConn := GetPgx(c)
+
+	rows, err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if err != nil {
+		return results, err
+	}
+
+	results, err = pgx.CollectRows(rows, pgx.RowToStructByName[T])
+	if err != nil {
+		return results, err
+	}
+
+	return results, nil
+}
+
+func Setup() (*pgx.Conn, *redis.Conn, *fiber.App, error) {
```

Comment:
It would be nice to create a struct to return all the variables.


```go
type Config struct {
    Postgress: pgx.Connnection
    Redis: redis.Connection
    App: fiber.App
}
```